File: backend/internal/constants/report.go
package constants

import "time"

// AbnormalLevel 异常等级枚举（README 枚举出现位置清单必列）。
const (
	AbnormalMild     = "mild"     // 轻度异常
	AbnormalModerate = "moderate" // 中度异常
	AbnormalSevere   = "severe"   // 重度异常
)

// AbnormalLevels 全部异常等级。
var AbnormalLevels = []string{AbnormalMild, AbnormalModerate, AbnormalSevere}

// ReportStatus 报告状态枚举（README 枚举出现位置清单必列）。
const (
	ReportDraft         = "draft"          // 草稿
	ReportGenerated     = "generated"      // 已生成
	ReportReviewed      = "reviewed"       // 已审核
	ReportPublished     = "published"      // 已发布
	ReportWithdrawing   = "withdrawing"    // 撤回审批中（冻结下载与新版本生成）
	ReportResignPending = "resign_pending" // 撤回已批准，待重签
	ReportWithdrawn     = "withdrawn"      // 已撤回归档（原版保留）
)

// ReportStatuses 全部报告状态。
var ReportStatuses = []string{
	ReportDraft, ReportGenerated, ReportReviewed, ReportPublished,
	ReportWithdrawing, ReportResignPending, ReportWithdrawn,
}

// WithdrawStatus 撤回申请状态。
const (
	WithdrawPending  = "pending"  // 待审批
	WithdrawApproved = "approved" // 已批准
	WithdrawRejected = "rejected" // 已驳回
)

// WithdrawStatuses 全部撤回申请状态。
var WithdrawStatuses = []string{WithdrawPending, WithdrawApproved, WithdrawRejected}

// ReportWithdrawWindow 已发布报告允许申请撤回的时间窗口：24 小时。
const ReportWithdrawWindow = 24 * time.Hour

// ResultStatus 检查结果状态。
const (
	ResultPending  = "pending"  // 待录入
	ResultEntered  = "entered"  // 已录入
	ResultReviewed = "reviewed" // 已审核
)

// RegistrationStatus 登记状态。
const (
	RegistrationRegistered = "registered"  // 已登记
	RegistrationInProgress = "in_progress" // 进行中
	RegistrationCompleted  = "completed"   // 已完成
)

// PackageStatus 套餐状态。
const (
	PackageActive   = "active"
	PackageInactive = "inactive"
)

// PackageType 套餐类型。
const (
	PackageEntry   = "entry"   // 入职体检
	PackageAnnual  = "annual"  // 年度体检
	PackagePremium = "premium" // 高端体检
	PackageOther   = "other"   // 其他
)

// GroupOrderStatus 团检订单状态。
const (
	GroupOrderPending   = "pending"
	GroupOrderConfirmed = "confirmed"
	GroupOrderDone      = "done"
)

// FollowUpStatus 复查跟踪状态。
const (
	FollowUpPending = "pending"
	FollowUpDone    = "done"
)

File: frontend/src/constants/report.ts
// 与 backend/internal/constants/report.go 保持一致
export const AbnormalLevel = {
  MILD: 'mild',
  MODERATE: 'moderate',
  SEVERE: 'severe',
} as const;

export const AbnormalLevelLabels: Record<string, string> = {
  [AbnormalLevel.MILD]: '轻度异常',
  [AbnormalLevel.MODERATE]: '中度异常',
  [AbnormalLevel.SEVERE]: '重度异常',
};

export const ReportStatus = {
  DRAFT: 'draft',
  GENERATED: 'generated',
  REVIEWED: 'reviewed',
  PUBLISHED: 'published',
  WITHDRAWING: 'withdrawing',
  WITHDRAWN: 'withdrawn',
} as const;

export const ReportStatusLabels: Record<string, string> = {
  [ReportStatus.DRAFT]: '草稿',
  [ReportStatus.GENERATED]: '已生成',
  [ReportStatus.REVIEWED]: '已审核',
  [ReportStatus.PUBLISHED]: '已发布',
  [ReportStatus.WITHDRAWING]: '撤回审批中',
  [ReportStatus.WITHDRAWN]: '已撤回（待重签）',
};

export const ReportWithdrawStatus = {
  PENDING: 'pending',
  APPROVED: 'approved',
  REJECTED: 'rejected',
} as const;

export const ReportWithdrawStatusLabels: Record<string, string> = {
  [ReportWithdrawStatus.PENDING]: '待审批',
  [ReportWithdrawStatus.APPROVED]: '已批准',
  [ReportWithdrawStatus.REJECTED]: '已驳回',
};

// 撤回申请时限（小时），与后端 constants.WithdrawWindowHours 保持一致。
export const WITHDRAW_WINDOW_HOURS = 24;

export const ResultStatusLabels: Record<string, string> = {
  pending: '待录入',
  entered: '已录入',
  reviewed: '已审核',
};

export const RegistrationStatusLabels: Record<string, string> = {
  registered: '已登记',
  in_progress: '进行中',
  completed: '已完成',
};

export const PackageStatusLabels: Record<string, string> = {
  active: '启用',
  inactive: '停用',
};

export const PackageTypeLabels: Record<string, string> = {
  entry: '入职体检',
  annual: '年度体检',
  premium: '高端体检',
  other: '其他',
};

export const FollowUpStatusLabels: Record<string, string> = {
  pending: '待复查',
  done: '已复查',
};
